test/server.js

```diff
@@ -13,9 +13,43 @@ const config = {
     port: 5432,
     name: 'sensorthingstest',
     user: 'postgres',
-    pass: '12345678'
+    pass: '12345678',
+    hooks: {
+      'beforeValidate': instance => {
+        instance.beforeValidate = 'beforeValidate';
+      },
+      'afterValidate': instance => {
+        instance.afterValidate = 'afterValidate';
+      },
+      'beforeCreate': instance => {
+        instance.beforeCreate = 'beforeCreate';
+      },
+      'afterCreate': instance => {
+        instance.afterCreate = 'afterCreate';
+      },
+      // From Sequelize documentation:
+      // (http://docs.sequelizejs.com/en/latest/docs/hooks/)
+      // "The only way to call beforeDestroy/afterDestroy hooks are on
+      // associations with `onDelete: 'cascade'` and the option `hooks: true`"
+      // TBD It seems our ST API implementation does not exercise these hooks.
+      // Perhaps we should not implement them.
+/*
+      'beforeDestroy': () => {
+      },
+      'afterDestroy': () => {
+      },
+*/
+      'beforeUpdate': (instance) => {
+        // TODO Updating an instance does not return the instance. Therefore,
+        // it still needs to be determined how to test this hook.
+      },
+      'afterUpdate': (instance) => {
+        // TODO Updating an instance does not return the instance. Therefore,
+        // it still needs to be determined how to test this hook.
+      }
```

Comment:
One thing you could do here is to keep an object where you add the hooks that has been executed as properties. Something like:

```javascript
let hooks = {};
const config = {
  db: {
    [...]
    hooks: {
      beforeCreate: () => { hooks[beforeCreate] = true },
      afterCreate: () => { hooks[afterCreate] = true }
    }
  }
}
```

Then you can export two functions (I'm sure you'll find better names ;)):

```javascript
export const resetHooks = () => { hooks = {} };
export const hooked = name => { return hook[name] !== undefined };
```

So before each test, you can call `resetHooks` and during the test you can check if the hook was executed or not.